src/client_reqwest.rs

```diff
@@ -102,44 +103,59 @@ impl AsyncTelegramApi for AsyncApi {
-                    Value::String(val) => val.to_string(),
-                    other => other.to_string(),
-                };
-
-                form = form.text(key.clone(), val);
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+            use reqwest::multipart;
+            use serde_json::Value;
+
+            let json_string = crate::json::encode(&params)?;
+            let json_struct: Value = serde_json::from_str(&json_string).unwrap();
+
+            let file_keys: Vec<&str> = files.iter().map(|(key, _)| *key).collect();
+            let files_with_paths: Vec<(String, &str, String)> = files
+                .iter()
+                .map(|(key, path)| {
+                    (
+                        (*key).to_string(),
+                        path.to_str().unwrap(),
+                        path.file_name().unwrap().to_str().unwrap().to_string(),
+                    )
+                })
+                .collect();
+
+            let mut form = multipart::Form::new();
+            for (key, val) in json_struct.as_object().unwrap() {
+                if !file_keys.contains(&key.as_str()) {
+                    let val = match val {
+                        Value::String(val) => val.to_string(),
+                        other => other.to_string(),
+                    };
+
+                    form = form.text(key.clone(), val);
+                }
             }
-        }
 
-        for (parameter_name, file_path, file_name) in files_with_paths {
-            let file = File::open(file_path)
-                .await
-                .map_err(|error| Error::Encode(error.to_string()))?;
-            let part = multipart::Part::stream(file).file_name(file_name);
-            form = form.part(parameter_name, part);
-        }
+            for (parameter_name, file_path, file_name) in files_with_paths {
+                let file = tokio::fs::File::open(file_path)
+                    .await
+                    .map_err(|error| Error::Encode(error.to_string()))?;
+                let part = multipart::Part::stream(file).file_name(file_name);
+                form = form.part(parameter_name, part);
+            }
 
-        let url = format!("{}/{method}", self.api_url);
+            let url = format!("{}/{method}", self.api_url);
 
-        let response = self.client.post(url).multipart(form).send().await?;
-        Self::decode_response(response).await
+            let response = self.client.post(url).multipart(form).send().await?;
+            Self::decode_response(response).await
+        }
+
+        #[cfg(target_arch = "wasm32")]
+        {
+            // for this function need api about fs, so we this function need find replacement,maybe another function input encoded file? welcome to PR.
+            Err(Error::Encode(format!(
+                "calling {method:?} with files is unsupported in WASM due to missing form_data / attachment support. Was called with params {params:?} and files {files:?}",
+            )))
```

Comment:
I think I would remove the comment. The explanation is already there two lines below and the comment doesn't add much. Improving the error message seems better than keeping that comment. Also, I would add 'currently' to the error message as thats something hopefully fixable and not completely impossible.

```suggestion
            Err(Error::Encode(format!(
                "calling {method:?} with files is currently unsupported in WASM due to missing form_data / attachment support. Was called with params {params:?} and files {files:?}",
            )))
```